# Supply Chain Finance (SCF) Opportunity Analysis

### Assessing District-Level Supply Chain Finance Opportunities Across Pakistan

**Prepared by:** Zujaja

**Tools Used:** Python, Pandas, NumPy, Matplotlib, Seaborn

# 📊 Dataset Overview: Supply Chain Finance Opportunity Data

## 1. Description

This project evaluates **Supply Chain Finance (SCF) opportunities across Pakistani districts** by integrating SME, supply-chain, infrastructure, banking, financing etc.

The analysis identifies districts and dominant industries with **higher SCF potential**, assesses financing needs, and maps suitable banking products to local supply-chain characteristics.

The final dataset supports **district-level opportunity analysis and Power BI visualization** for strategic banking decisions.


---

## 2. Dataset Structure & Key Dimensions

The analysis uses **three worksheets** containing complementary district-level information:

| **Sheet**   | **Focus Area**                      | **Key Information**                                                                                       |
| ----------- | ----------------------------------- | --------------------------------------------------------------------------------------------------------- |
| **Sheet 1** | Supply Chain Data     | SME activity, anchor sectors, employment, exports, banking penetration, value chains                      |
| **Sheet 2** | Ease of Doing Business | Infrastructure, credit access, business environment, logistics, production, processing and trade activity |
| **Sheet 3** | Import & Export Data            | Major export products, destinations, import products and import sources                                   |

The three worksheets are combined using **District and Province** as the primary geographic identifiers to create an integrated SCF opportunity dataset.

---

# 3. Data Dictionary

## Sheet 1 — Supply Chain Data 

This worksheet captures the **SME base, anchor-sector presence, employment, export orientation, banking penetration, and dominant value chains** across districts.

| **#** | **Column Name**                   | **Data Type** | **Description**                                                   |
| ----: | --------------------------------- | ------------- | ----------------------------------------------------------------- |
|     1 | **Year**                          | `int64`       | Reporting year                                                    |
|     2 | **Province**                      | `object`      | Province of the district                                          |
|     3 | **District**                      | `object`      | District-level geographic identifier                              |
|     4 | **Key_Anchor_Sectors**            | `object`      | Major sectors or industries supporting local supply chains        |
|     5 | **Anchor_firm_presence**          | `object`      | Presence of major anchor firms within the district                |
|     6 | **Avg_Employees_per_SME**         | `float64`     | Average number of employees per SME                               |
|     7 | **Export_Oriented**               | `object`      | Indicates the district's export orientation                       |
|     8 | **Estimated_Number_of_SMEs**      | `float64`     | Estimated number of SMEs operating in the district                |
|     9 | **Estimated_Number_of_Employees** | `float64`     | Estimated employment generated by SMEs                            |
|    10 | **Banking_Penetration (%)**       | `float64`     | Estimated level of banking penetration                            |
|    11 | **SME_Opportunity_Score**         | `float64`     | Score representing the overall SME opportunity potential          |
|    12 | **Dominant_Value_Chain**          | `object`      | Main value chain associated with the district's economic activity |

---

## Sheet 2 — Ease of Doing Business

This worksheet measures the **operating environment and supply-chain conditions** that influence financing requirements and SCF feasibility.

| **#** | **Column Name**                                   | **Data Type** | **Description**                                           |
| ----: | ------------------------------------------------- | ------------- | --------------------------------------------------------- |
|     1 | **District**                                      | `object`      | District-level geographic identifier                      |
|     2 | **Province**                                      | `object`      | Province of the district                                  |
|     3 | **Dominant_Industry**                             | `object`      | Primary industry driving district-level economic activity |
|     4 | **Infrastructure (Roads, Electricity, Internet)** | `object`      | Availability and quality of key infrastructure            |
|     5 | **Banking_Access**                                | `object`      | Accessibility of formal banking services                  |
|     6 | **Access_to_Credit / Lending**                    | `object`      | Availability of credit and lending facilities             |
|     7 | **Business_Registration**                         | `object`      | Ease and accessibility of business registration           |
|     8 | **Tax_System**                                    | `object`      | Conditions associated with the local tax system           |
|     9 | **Property_Registration**                         | `object`      | Ease of property registration                             |
|    10 | **Contract_Enforcement**                          | `object`      | Effectiveness of contract enforcement mechanisms          |
|    11 | **Skilled_Labor_Availability**                    | `object`      | Availability of skilled workforce                         |
|    12 | **Export & Trade_Facilitation**                   | `object`      | Availability of trade and export facilitation mechanisms  |
|    13 | **Digital_payment_Adoption**                      | `object`      | Level of digital payment adoption                         |
|    14 | **Logistics/Transport_Connectivity**              | `object`      | Quality of transport and logistics connectivity           |
|    15 | **Supply_Chain_Constraints**                      | `object`      | Major constraints affecting local supply chains           |
|    16 | **Urbanization_Level**                            | `object`      | Level of urbanization within the district                 |
|    17 | **Production_Volume**                             | `float64`     | Estimated production volume                               |
|    18 | **Processing_Capacity**                           | `float64`     | Estimated capacity for processing local production        |
|    19 | **Export_Activity**                               | `object`      | Level or presence of export activity                      |
|    20 | **Import_Activity**                               | `object`      | Level or presence of import activity                      |

---

## Sheet 3 — Import & Export Data
This worksheet captures the district-level **trade profile**, including major products traded internationally and key trading markets.

| **#** | **Column Name**             | **Data Type** | **Description**                                       |
| ----: | --------------------------- | ------------- | ----------------------------------------------------- |
|     1 | **District**                | `object`      | District-level geographic identifier                  |
|     2 | **Province**                | `object`      | Province of the district                              |
|     3 | **Major Export Products**   | `object`      | Major products exported from the district             |
|     4 | **Key Export Destinations** | `object`      | Major international destinations for district exports |
|     5 | **Major Import Products**   | `object`      | Major products imported into the district             |
|     6 | **Key Import Sources**      | `object`      | Major countries or markets supplying imports          |

---

# 4. Data Preparation & Analysis Workflow

The worksheets were cleaned, integrated, analysed, and transformed using Python to create a **ready-to-visualize SCF opportunity dataset** for Power BI.

The analysis follows a structured workflow:

1. Import required libraries.
2. Load the three source worksheets.
3. Inspect datasets, dimensions, data types, null values, and duplicates.
4. Clean and standardize district, province, industry, and categorical values.
5. Develop and standardize analytical indicators.
6. Calculate Supply Chain Finance opportunity components.
7. Generate the final SCF Opportunity Score.
8. Classify districts into Low, Medium, and High opportunity levels.
9. Map district characteristics to recommended banking products.
10. Validate the final dataset and analytical outputs.
11. Export the processed dataset as a CSV file for Power BI visualization.


---


## Import Required Libraries

Import essential libraries for **data manipulation, analysis, and visualization**.


In [1]:
# Data manipulation
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns


# Ignore unnecessary warnings
import warnings
warnings.filterwarnings('ignore')

print("Libraries loaded successfully!")

Libraries loaded successfully!


### Outcome

Required libraries are successfully loaded and ready for analysis.


## Load Source Workbook

Load the Excel workbook and identify the available worksheets for analysis.


In [2]:
# Import Excel file
file_path = "Supply chain data workbook.xlsx"

# Read all sheets
excel_data = pd.ExcelFile(file_path)

# Check sheet names
excel_data.sheet_names

['Ease of Doing Business', 'Import and Export', 'Supply Chain Data']

### Outcome

The workbook is successfully loaded, and the available **three source worksheets** are identified.


## Load Supply Chain Data

Load the **Supply Chain Data** worksheet into a Pandas DataFrame for analysis.


In [3]:
df1 = pd.read_excel(
    file_path,
    sheet_name="Supply Chain Data"
)

df1

,Year,Province,District,Key_Anchor_Sectors,Anchor_firm_presence,Avg_Employees_per_SME,Export_Oriented,Estimated_Number_of_SMEs,Estimated_Number_of_Employees,Banking_Penetration (%),SME Opportunity_Score,Dominant_Value_Chain
0,2016,Khyber Pakhtunkhwa,Abbottabad,Hospitality & Tourism,Yes,9,No,7859,70731,55.0,22.16,Cotton
1,2017,Khyber Pakhtunkhwa,Abbottabad,Hospitality & Tourism,Yes,9,No,7941,71469,55.0,22.21,Cotton
2,2018,Khyber Pakhtunkhwa,Abbottabad,Hospitality & Tourism,Yes,10,No,8024,80240,55.0,22.41,Cotton
3,2019,Khyber Pakhtunkhwa,Abbottabad,Hospitality & Tourism,Yes,10,No,8108,81080,55.0,22.46,Cotton
4,2020,Khyber Pakhtunkhwa,Abbottabad,Hospitality & Tourism,Yes,9,No,8193,68575,55.0,22.36,Cotton
...,...,...,...,...,...,...,...,...,...,...,...,...
405,2021,Balochistan,Zhob,Livestock & Dairy,No,7,No,2025,13892,25.0,8.28,Livestock & Dairy
406,2022,Balochistan,Zhob,Livestock & Dairy,No,8,No,2071,16568,25.0,8.35,Livestock & Dairy
407,2023,Balochistan,Zhob,Livestock & Dairy,No,8,No,2119,17291,25.0,8.38,Livestock & Dairy
408,2024,Balochistan,Zhob,Livestock & Dairy,No,8,No,2168,18038,25.0,8.41,Livestock & Dairy


### Outcome

The **Supply Chain Data** worksheet is successfully loaded into `df1`.


## Load Ease of Doing Business Data

Load the **Ease of Doing Business** worksheet into a Pandas DataFrame for analysis.


In [4]:
df2 = pd.read_excel(
    file_path,
    sheet_name="Ease of Doing Business"
)

df2.head()

,District,Province,Dominant_Industry,"Infrastructure (Roads, Electricity, Internet)",Banking_Access,Access_to_Credit / Lending,Business_Registration,Tax_System,Property_Registration,Contract_Enforcement,...,Skilled_Labor_Availability,Export & Trade_Facilitation,Digital_payment_Adoption,Logistics/Transport_Connectivity,Supply_Chain_Constraints,Urbanization_Level,Production_Volume,Processing_Capacity,Export_Activity,Import_Activity
0,Islamabad,Punjab,Professional Services,High,High,High,High,High,High,High,...,High,Medium,High,High,Low,High,Medium,Medium,High,High
1,Lahore,Punjab,Retail & Wholesale Trade,High,High,High,High,High,High,High,...,High,High,High,High,Low,High,High,High,High,High
2,Faisalabad,Punjab,Textile & Apparel,High,High,High,Medium,Medium,Medium,Medium,...,High,High,High,High,Medium,High,High,High,High,High
3,Rawalpindi,Punjab,Professional Services,High,High,High,High,High,High,High,...,High,Medium,High,High,Low,High,High,High,High,High
4,Multan,Punjab,Agriculture & Agribusiness,Medium,High,Medium,Medium,Medium,Medium,Medium,...,High,Medium,High,High,Medium,High,High,High,Medium,Medium


### Outcome

The **Ease of Doing Business** worksheet is successfully loaded into `df2`.


## Load Import and Export Data

Load the **Import and Export** worksheet into a Pandas DataFrame for analysis.


In [5]:
df3 = pd.read_excel(
    file_path,
    sheet_name="Import and Export"
)

df3

,District,Province,Major Export Products,Key_Export_Destinations,Major_Import_Products,Key_Import_Sources
0,Islamabad,Punjab,IT & Digital Business Services,"UAE, Afghanistan, China, UK",Industrial Machinery & Materials,"China, UAE, Saudi Arabia, USA"
1,Lahore,Punjab,Textiles & Industrial Goods,"USA, UK, Germany, Netherlands, Italy",Industrial Machinery & Materials,"China, UAE, USA, South Korea"
2,Faisalabad,Punjab,Textiles & Industrial Goods,"USA, UK, UAE, Canada",Industrial Machinery & Materials,"USA, China, UAE, Brazil"
3,Rawalpindi,Punjab,IT & Digital Business Services,"UAE, China, Afghanistan, Saudi Arabia",Industrial Machinery & Materials,"China, UAE, Saudi Arabia, USA"
4,Multan,Punjab,Agri-Commodity & Food Products,"USA, UAE, UK, Germany",Industrial Machinery & Materials,"China, UAE, Saudi Arabia, USA"
5,Gujranwala,Punjab,Engineering & Industrial Products,"USA, Germany, UK, France, Netherlands",Industrial Machinery & Materials,"China, UAE, USA, Germany"
6,Sialkot,Punjab,"Sports, Surgical & Leather Products","UAE, China, Saudi Arabia, Afghanistan",Industrial Machinery & Materials,"China, UAE, South Korea, Japan"
7,Sargodha,Punjab,Agri-Commodity & Food Products,"UAE, China, Afghanistan, Saudi Arabia",Agricultural Inputs & Machinery,"China, Saudi Arabia, UAE, USA"
8,Bahawalpur,Punjab,Agri-Commodity & Food Products,"UAE, Saudi Arabia, China, Afghanistan",Agricultural Inputs & Machinery,"China, UAE, Saudi Arabia, Malaysia"
9,Sahiwal,Punjab,Agri-Commodity & Food Products,"UAE, Saudi Arabia, China, Afghanistan",Agricultural Inputs & Machinery,"China, Malaysia, UAE, Indonesia"


### Outcome

The **Import and Export** worksheet is successfully loaded into `df3`.


## Initial Data Quality Check

Create a summary of the three DataFrames to check their **size, missing values, and duplicate records**.


In [6]:
summary = pd.DataFrame({
    "DataFrames": ["df1", "df2", "df3"],
    "Rows": [len(df1), len(df2), len(df3)],
    "Columns": [df1.shape[1], df2.shape[1], df3.shape[1]],
    "Total_Nulls": [
        df1.isnull().sum().sum(),
        df2.isnull().sum().sum(),
        df3.isnull().sum().sum()
    ],
    "Duplicate_Rows": [
        df1.duplicated().sum(),
        df2.duplicated().sum(),
        df3.duplicated().sum()
    ]
})

summary

,DataFrames,Rows,Columns,Total_Nulls,Duplicate_Rows
0,df1,410,12,0,0
1,df2,41,21,0,0
2,df3,41,6,0,0


### Outcome

All three datasets contain **no missing values and no duplicate rows**. The data is complete and ready for the next stage of cleaning and analysis.


## Inspect Column Structure

Review the column names of each DataFrame to **understand the available variables and ensure consistent data structure** before merging and analysis.


In [7]:
for name, df in {"df1": df1, "df2": df2, "df3": df3}.items():
    print(f"\n{'='*60}")
    print(name)
    print(f"{'='*60}")
    print(df.columns.tolist())


df1
['Year', 'Province', 'District', 'Key_Anchor_Sectors', 'Anchor_firm_presence', 'Avg_Employees_per_SME', 'Export_Oriented', 'Estimated_Number_of_SMEs', 'Estimated_Number_of_Employees', 'Banking_Penetration (%)', 'SME Opportunity_Score', 'Dominant_Value_Chain']

df2
['District', 'Province', 'Dominant_Industry', 'Infrastructure (Roads, Electricity, Internet)', 'Banking_Access', 'Access_to_Credit / Lending', 'Business_Registration', 'Tax_System', 'Property_Registration', 'Contract_Enforcement', 'Legal & Regulatory_Environment', 'Skilled_Labor_Availability', 'Export & Trade_Facilitation', 'Digital_payment_Adoption', 'Logistics/Transport_Connectivity', 'Supply_Chain_Constraints', 'Urbanization_Level', 'Production_Volume', 'Processing_Capacity', 'Export_Activity', 'Import_Activity']

df3
['District', 'Province', 'Major Export Products', 'Key_Export_Destinations', 'Major_Import_Products', 'Key_Import_Sources']


### Outcome

The three datasets contain **complementary SME, business environment, supply-chain, and trade variables** required for the SCF opportunity analysis.


## Check District Coverage

Compare the number of unique districts across all three datasets to verify **consistent geographic coverage**.


In [8]:
# Check unique district counts
print("df1 districts:", df1["District"].nunique())
print("df2 districts:", df2["District"].nunique())
print("df3 districts:", df3["District"].nunique())


df1 districts: 41
df2 districts: 41
df3 districts: 41


### Outcome

All three datasets cover **41 unique districts**, confirming consistent district-level geographic coverage.


## Standardize and Merge Data

Standardize district and province names, check duplicate districts, and merge the SME and supply-chain datasets using **District and Province** as common identifiers.


In [9]:
for df in [df1, df2, df3]:
    df["District"] = df["District"].astype(str).str.strip()
    df["Province"] = df["Province"].astype(str).str.strip()

In [10]:
print("df2 duplicate Districts:", df2["District"].duplicated().sum())
print("df3 duplicate Districts:", df3["District"].duplicated().sum())

df2 duplicate Districts: 0
df3 duplicate Districts: 0


In [11]:
df_final = df1.merge(
    df2,
    on=["District", "Province"],
    how="left"
)

In [12]:
print("Final Shape:", df_final.shape)

df_final.head()

Final Shape: (410, 31)


,Year,Province,District,Key_Anchor_Sectors,Anchor_firm_presence,Avg_Employees_per_SME,Export_Oriented,Estimated_Number_of_SMEs,Estimated_Number_of_Employees,Banking_Penetration (%),...,Skilled_Labor_Availability,Export & Trade_Facilitation,Digital_payment_Adoption,Logistics/Transport_Connectivity,Supply_Chain_Constraints,Urbanization_Level,Production_Volume,Processing_Capacity,Export_Activity,Import_Activity
0,2016,Khyber Pakhtunkhwa,Abbottabad,Hospitality & Tourism,Yes,9,No,7859,70731,55.0,...,High,Low,High,High,Medium,Medium,Medium,High,Low,Medium
1,2017,Khyber Pakhtunkhwa,Abbottabad,Hospitality & Tourism,Yes,9,No,7941,71469,55.0,...,High,Low,High,High,Medium,Medium,Medium,High,Low,Medium
2,2018,Khyber Pakhtunkhwa,Abbottabad,Hospitality & Tourism,Yes,10,No,8024,80240,55.0,...,High,Low,High,High,Medium,Medium,Medium,High,Low,Medium
3,2019,Khyber Pakhtunkhwa,Abbottabad,Hospitality & Tourism,Yes,10,No,8108,81080,55.0,...,High,Low,High,High,Medium,Medium,Medium,High,Low,Medium
4,2020,Khyber Pakhtunkhwa,Abbottabad,Hospitality & Tourism,Yes,9,No,8193,68575,55.0,...,High,Low,High,High,Medium,Medium,Medium,High,Low,Medium


### Outcome

The datasets are standardized and successfully merged into a **single district-level dataset (`df_final`)** for further analysis.


## Inspect Final Dataset

Review the structure, data types, and completeness of the merged dataset before further analysis.


In [13]:
df_final.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 410 entries, 0 to 409
Data columns (total 31 columns):
 #   Column                                         Non-Null Count  Dtype  
---  ------                                         --------------  -----  
 0   Year                                           410 non-null    int64  
 1   Province                                       410 non-null    object 
 2   District                                       410 non-null    object 
 3   Key_Anchor_Sectors                             410 non-null    object 
 4   Anchor_firm_presence                           410 non-null    object 
 5   Avg_Employees_per_SME                          410 non-null    int64  
 6   Export_Oriented                                410 non-null    object 
 7   Estimated_Number_of_SMEs                       410 non-null    int64  
 8   Estimated_Number_of_Employees                  410 non-null    int64  
 9   Banking_Penetration (%)                        410 non

### Outcome

The merged dataset contains **410 records and 31 columns**, with all fields populated and no missing values.


# Feature Engineering

Feature engineering was performed to transform qualitative variables into numerical values and create new analytical features. This step prepares the dataset for scoring, comparison, and further analysis.

## SME Scale Score

Classify districts into **three SME scale levels** based on the estimated number of SMEs. This scoring helps identify districts with a **relatively larger SME base**, which may indicate greater potential for supply chain financing.


In [14]:
# SME Scale Score
df_final["SME_Scale_Score"] = pd.qcut(
    df_final["Estimated_Number_of_SMEs"].rank(method="first"),
    q=3,
    labels=[1, 2, 3]
).astype(int)

df_final[["District", "Estimated_Number_of_SMEs", "SME_Scale_Score"]].head()

,District,Estimated_Number_of_SMEs,SME_Scale_Score
0,Abbottabad,7859,1
1,Abbottabad,7941,1
2,Abbottabad,8024,1
3,Abbottabad,8108,1
4,Abbottabad,8193,1


### Outcome

Each district is assigned an **SME Scale Score from 1 to 3**, representing its relative SME concentration and supporting the assessment of **district-level SCF potential**.


## Employment Scale Score

Classify districts into **three employment scale levels** based on estimated SME employment. This helps identify districts with a **larger workforce base**, indicating greater economic and potential financing activity.


In [15]:
# Employment Scale Score
df_final["Employment_Scale_Score"] = pd.qcut(
    df_final["Estimated_Number_of_Employees"].rank(method="first"),
    q=3,
    labels=[1, 2, 3]
).astype(int)

df_final[[
    "District",
    "Estimated_Number_of_Employees",
    "Employment_Scale_Score"
]].head()

,District,Estimated_Number_of_Employees,Employment_Scale_Score
0,Abbottabad,70731,1
1,Abbottabad,71469,1
2,Abbottabad,80240,1
3,Abbottabad,81080,1
4,Abbottabad,68575,1


### Outcome

Each district is assigned an **Employment Scale Score from 1 to 3**, reflecting its relative SME employment size.


## Anchor Firm Score

Convert anchor-firm presence into a **binary score** to identify districts connected to established businesses that can support structured supply-chain financing.


In [16]:
# Anchor Firm Score
df_final["Anchor_Firm_Score"] = (
    df_final["Anchor_firm_presence"]
    .astype(str)
    .str.strip()
    .str.lower()
    .map({
        "yes": 1,
        "no": 0
    })
    .fillna(0)
)

df_final[["District", "Anchor_firm_presence", "Anchor_Firm_Score"]].head()

,District,Anchor_firm_presence,Anchor_Firm_Score
0,Abbottabad,Yes,1
1,Abbottabad,Yes,1
2,Abbottabad,Yes,1
3,Abbottabad,Yes,1
4,Abbottabad,Yes,1


### Outcome

Each district is assigned an **Anchor Firm Score of 1 or 0**, based on the presence or absence of an anchor firm.


## Production Capacity Score

Convert production volume into a **three-level score** to measure the scale of local production. Higher production capacity indicates **greater supply-chain activity and potential financing demand**.


In [17]:
# Production Capacity Score
df_final["Production_Capacity_Score"] = (
    df_final["Production_Volume"]
    .astype(str)
    .str.strip()
    .str.lower()
    .map({
        "high": 3,
        "medium": 2,
        "low": 1
    })
    .fillna(1)
)

df_final[[
    "District",
    "Production_Volume",
    "Production_Capacity_Score"
]].head()

,District,Production_Volume,Production_Capacity_Score
0,Abbottabad,Medium,2
1,Abbottabad,Medium,2
2,Abbottabad,Medium,2
3,Abbottabad,Medium,2
4,Abbottabad,Medium,2


### Outcome

Each district is assigned a **Production Capacity Score from 1 to 3**, based on its production volume level.


## Processing Capacity Score

Convert processing capacity into a **three-level score** to measure the district's ability to process local production and support supply-chain activity.


In [18]:
# Processing Capacity Score
df_final["Processing_Capacity_Score"] = (
    df_final["Processing_Capacity"]
    .astype(str)
    .str.strip()
    .str.lower()
    .map({
        "high": 3,
        "medium": 2,
        "low": 1
    })
    .fillna(1)
)

df_final[[
    "District",
    "Processing_Capacity",
    "Processing_Capacity_Score"
]].head()

,District,Processing_Capacity,Processing_Capacity_Score
0,Abbottabad,High,3
1,Abbottabad,High,3
2,Abbottabad,High,3
3,Abbottabad,High,3
4,Abbottabad,High,3


### Outcome

Each district is assigned a **Processing Capacity Score from 1 to 3**, based on its processing capacity level.


## Logistics Score

Convert logistics and transport connectivity into a **three-level score** to assess how effectively goods can move through the supply chain. Better connectivity can support smoother trade and financing activities.


In [32]:
# Logistics Score
df_final["Logistics_Score"] = (
    df_final["Logistics/Transport_Connectivity"]
    .astype(str)
    .str.strip()
    .str.lower()
    .map({
        "high": 3,
        "medium": 2,
        "low": 1
    })
    .fillna(1)
)

df_final[[
    "District",
    "Logistics/Transport_Connectivity",
    "Logistics_Score"
]].head()

,District,Logistics/Transport_Connectivity,Logistics_Score
0,Abbottabad,High,3
1,Abbottabad,High,3
2,Abbottabad,High,3
3,Abbottabad,High,3
4,Abbottabad,High,3


### Outcome

Each district is assigned a **Logistics Score from 1 to 3**, based on its transport and connectivity level.


## Credit Access Score

Convert access to credit into a **three-level score** to identify districts with greater financing gaps. A higher score indicates **lower access to credit and therefore potentially higher financing need**.


In [33]:
# Credit Access Score
df_final["Credit_Access_Score"] = (
    df_final["Access_to_Credit / Lending"]
    .astype(str)
    .str.strip()
    .str.lower()
    .map({
        "high": 1,
        "medium": 2,
        "low": 3
    })
    .fillna(2)
)

df_final[[
    "District",
    "Access_to_Credit / Lending",
    "Credit_Access_Score"
]].head()

,District,Access_to_Credit / Lending,Credit_Access_Score
0,Abbottabad,Medium,2
1,Abbottabad,Medium,2
2,Abbottabad,Medium,2
3,Abbottabad,Medium,2
4,Abbottabad,Medium,2


### Outcome

Each district is assigned a **Credit Access Score from 1 to 3**, with higher scores representing greater potential financing need.


## Banking Access Score

Convert banking access into a **three-level score** to identify districts with limited access to formal banking services. Lower banking access results in a **higher potential financing need**.


In [34]:
# Banking Access Score
df_final["Banking_Access_Score"] = (
    df_final["Banking_Access"]
    .astype(str)
    .str.strip()
    .str.lower()
    .map({
        "high": 1,
        "medium": 2,
        "low": 3
    })
    .fillna(2)
)

df_final[[
    "District",
    "Banking_Access",
    "Banking_Access_Score"
]].head()

,District,Banking_Access,Banking_Access_Score
0,Abbottabad,Medium,2
1,Abbottabad,Medium,2
2,Abbottabad,Medium,2
3,Abbottabad,Medium,2
4,Abbottabad,Medium,2


### Outcome

Each district is assigned a **Banking Access Score from 1 to 3**, with higher scores indicating lower banking access and greater potential financing need.


## Supply Chain Constraint Score

Convert supply-chain constraints into a **three-level score** to measure the severity of operational barriers. Higher constraints indicate **greater supply-chain financing needs**.


In [35]:
# Supply Chain Constraint Score
df_final["Supply_Chain_Constraint_Score"] = (
    df_final["Supply_Chain_Constraints"]
    .astype(str)
    .str.strip()
    .str.lower()
    .map({
        "high": 3,
        "medium": 2,
        "low": 1
    })
    .fillna(2)
)

df_final[[
    "District",
    "Supply_Chain_Constraints",
    "Supply_Chain_Constraint_Score"
]].head()

,District,Supply_Chain_Constraints,Supply_Chain_Constraint_Score
0,Abbottabad,Medium,2
1,Abbottabad,Medium,2
2,Abbottabad,Medium,2
3,Abbottabad,Medium,2
4,Abbottabad,Medium,2


### Outcome

Each district is assigned a **Supply Chain Constraint Score from 1 to 3**, with higher scores representing greater supply-chain constraints.


## Working Capital Gap Score

Combine **credit access, production capacity, SME scale, and supply-chain constraints** into a single score to estimate the district's **working capital financing need**.


In [36]:
df_final["Working_Capital_Gap_Score"] = (
    df_final[
        [
            "Credit_Access_Score",
            "Production_Capacity_Score",
            "SME_Scale_Score",
            "Supply_Chain_Constraint_Score"
            
        ]
    ].mean(axis=1).round(2)
)

In [37]:
df_final[[
    "District",
    "Credit_Access_Score",
    "Production_Capacity_Score",
    "SME_Scale_Score",
    "Working_Capital_Gap_Score"
]].head()

,District,Credit_Access_Score,Production_Capacity_Score,SME_Scale_Score,Working_Capital_Gap_Score
0,Abbottabad,2,2,1,1.75
1,Abbottabad,2,2,1,1.75
2,Abbottabad,2,2,1,1.75
3,Abbottabad,2,2,1,1.75
4,Abbottabad,2,2,1,1.75


### Outcome

A **Working Capital Gap Score** is calculated for each district, where a higher score indicates greater potential working capital financing needs.


## Export Potential Score

Convert export orientation into a **binary score** to identify districts with export-linked supply chains and potential **trade finance requirements**.


In [38]:
# Export Potential Score
df_final["Export_Potential_Score"] = (
    df_final["Export_Oriented"]
    .astype(str)
    .str.strip()
    .str.lower()
    .map({
        "yes": 1,
        "no": 0,
    })
    .fillna(1)
)

df_final[[
    "District",
    "Export_Oriented",
    "Export_Potential_Score"
]].head()

,District,Export_Oriented,Export_Potential_Score
0,Abbottabad,No,0
1,Abbottabad,No,0
2,Abbottabad,No,0
3,Abbottabad,No,0
4,Abbottabad,No,0


### Outcome

Each district is assigned an **Export Potential Score of 1 or 0**, based on its export orientation.


## Import Potential Score

Convert import activity into a **binary score** to identify districts with import-linked supply chains and potential **trade financing requirements**.


In [39]:
# Import Potential Score
df_final["Import_Potential_Score"] = (
    df_final["Import_Activity"]
    .astype(str)
    .str.strip()
    .str.lower()
    .map({
        "yes": 1,
        "no": 0,
    })
    .fillna(1)
)

df_final[[
    "District",
    "Import_Activity",
    "Import_Potential_Score"
]].head()

,District,Import_Activity,Import_Potential_Score
0,Abbottabad,Medium,1.0
1,Abbottabad,Medium,1.0
2,Abbottabad,Medium,1.0
3,Abbottabad,Medium,1.0
4,Abbottabad,Medium,1.0


### Outcome

Each district is assigned an **Import Potential Score of 1 or 0**, based on its import activity.


## Export Activity Score

Convert export activity into a **three-level score** to measure the intensity of export activity and its potential for **trade and supply-chain financing**.


In [40]:
# Export Activity Score
df_final["Export_Activity_Score"] = (
    df_final["Export_Activity"]
    .astype(str)
    .str.strip()
    .str.lower()
    .map({
        "high": 3,
        "medium": 2,
        "low": 1,
    })
    .fillna(1)
)

df_final[[
    "District",
    "Export_Activity",
    "Export_Activity_Score"
]].head()

,District,Export_Activity,Export_Activity_Score
0,Abbottabad,Low,1
1,Abbottabad,Low,1
2,Abbottabad,Low,1
3,Abbottabad,Low,1
4,Abbottabad,Low,1


### Outcome

Each district is assigned an **Export Activity Score from 1 to 3**, based on its level of export activity.


## Import Activity Score

Convert import activity into a **three-level score** to measure the intensity of import activity and its potential for **trade and supply-chain financing**.


In [41]:
# Import Activity Score
df_final["Import_Activity_Score"] = (
    df_final["Import_Activity"]
    .astype(str)
    .str.strip()
    .str.lower()
    .map({
        "high": 3,
        "medium": 2,
        "low": 1,
        
    })
    .fillna(1)
)

df_final[[
    "District",
    "Import_Activity",
    "Import_Activity_Score"
]].head()

,District,Import_Activity,Import_Activity_Score
0,Abbottabad,Medium,2
1,Abbottabad,Medium,2
2,Abbottabad,Medium,2
3,Abbottabad,Medium,2
4,Abbottabad,Medium,2


### Outcome

Each district is assigned an **Import Activity Score from 1 to 3**, based on its level of import activity.


## Digital Payment Score

Convert digital payment adoption into a **three-level score** to measure district-level digital readiness for SCF.


In [42]:
# Digital Payment Score
df_final["Digital_Payment_Score"] = (
    df_final["Digital_payment_Adoption"]
    .astype(str)
    .str.strip()
    .str.lower()
    .map({
        "high": 3,
        "medium": 2,
        "low": 1
    })
    .fillna(1)
)

df_final[[
    "District",
    "Digital_payment_Adoption",
    "Digital_Payment_Score"
]].head()

,District,Digital_payment_Adoption,Digital_Payment_Score
0,Abbottabad,High,3
1,Abbottabad,High,3
2,Abbottabad,High,3
3,Abbottabad,High,3
4,Abbottabad,High,3


### Outcome

Each district is assigned a **Digital Payment Score from 1 to 3**, based on its level of digital payment adoption.


## Infrastructure Score

Convert infrastructure availability into a **three-level score** to assess the district's readiness for efficient supply-chain operations and SCF.


In [43]:
# Infrastructure Score
df_final["Infrastructure_Score"] = (
    df_final["Infrastructure (Roads, Electricity, Internet)"]
    .astype(str)
    .str.strip()
    .str.lower()
    .map({
        "high": 3,
        "medium": 2,
        "low": 1
    })
    .fillna(1)
)

df_final[[
    "District",
    "Infrastructure (Roads, Electricity, Internet)",
    "Infrastructure_Score"
]].head()

,District,"Infrastructure (Roads, Electricity, Internet)",Infrastructure_Score
0,Abbottabad,High,3.0
1,Abbottabad,High,3.0
2,Abbottabad,High,3.0
3,Abbottabad,High,3.0
4,Abbottabad,High,3.0


### Outcome

Each district is assigned an **Infrastructure Score from 1 to 3**, based on its infrastructure level.


## Business Environment Score

Convert key business-environment indicators into **three-level scores** and combine them into an overall score to assess the district's **business and operational readiness for SCF**.


In [44]:
# Convert business environment indicators to scores

business_cols = [
    "Business_Registration",
    "Tax_System",
    "Property_Registration",
    "Contract_Enforcement",
    "Legal & Regulatory_Environment",
    "Skilled_Labor_Availability",
    "Export & Trade_Facilitation"
]

score_map = {
    "high": 3,
    "medium": 2,
    "low": 1
}

for col in business_cols:
    df_final[col + "_Score"] = (
        df_final[col]
        .astype(str)
        .str.strip()
        .str.lower()
        .map(score_map)
        .fillna(1)
    )

df_final["Business_Environment_Score"] = (
    df_final[[col + "_Score" for col in business_cols]]
    .mean(axis=1)
    .round(2)
)

df_final[[
    "District",
    "Business_Environment_Score"
]].head()

,District,Business_Environment_Score
0,Abbottabad,2.14
1,Abbottabad,2.14
2,Abbottabad,2.14
3,Abbottabad,2.14
4,Abbottabad,2.14


### Outcome

A **Business Environment Score** is calculated for each district by averaging the selected indicators. Higher scores indicate a more supportive business environment.


## Supply Chain Potential Score

Combine SME scale, employment, anchor firms, production, processing capacity, and logistics to measure the district's **overall supply-chain potential**.


In [45]:
# Supply Chain Potential Score

supply_chain_cols = [
    "SME_Scale_Score",
    "Employment_Scale_Score",
    "Anchor_Firm_Score",
    "Production_Capacity_Score",
    "Processing_Capacity_Score",
    "Logistics_Score"
]

df_final["Supply_Chain_Potential_Score"] = (
    df_final[supply_chain_cols]
    .mean(axis=1)
    .round(2)
)

df_final[[
    "District",
    "Supply_Chain_Potential_Score"
]].head()

,District,Supply_Chain_Potential_Score
0,Abbottabad,1.83
1,Abbottabad,1.83
2,Abbottabad,1.83
3,Abbottabad,1.83
4,Abbottabad,1.83


### Outcome

A **Supply Chain Potential Score** is calculated for each district, with higher scores indicating stronger supply-chain potential.


## Financing Need Score

Combine credit access, banking access, working capital gap, and supply-chain constraints to measure the district's **overall financing need**.


In [46]:
# Financing Need Score

financing_cols = [
    "Credit_Access_Score",
    "Banking_Access_Score",
    "Working_Capital_Gap_Score",
    "Supply_Chain_Constraint_Score"
]

df_final["Financing_Need_Score"] = (
    df_final[financing_cols]
    .mean(axis=1)
    .round(2)
)

df_final[[
    "District",
    "Financing_Need_Score"
]].head()

,District,Financing_Need_Score
0,Abbottabad,1.94
1,Abbottabad,1.94
2,Abbottabad,1.94
3,Abbottabad,1.94
4,Abbottabad,1.94


### Outcome

A **Financing Need Score** is calculated for each district, with higher scores indicating **greater potential financing needs**.


## Digital Readiness Score

Combine digital payment adoption, infrastructure, and business environment indicators to measure the district's **overall digital and operational readiness for SCF**.


In [47]:
# Digital Readiness Score

df_final["Digital_Readiness_Score"] = (
    df_final[
        [
            "Digital_Payment_Score",
            "Infrastructure_Score",
            "Business_Environment_Score"
        ]
    ]
    .mean(axis=1)
    .round(2)
)

df_final[[
    "District",
    "Digital_Readiness_Score"
]].head()

,District,Digital_Readiness_Score
0,Abbottabad,2.71
1,Abbottabad,2.71
2,Abbottabad,2.71
3,Abbottabad,2.71
4,Abbottabad,2.71


### Outcome

A **Digital Readiness Score** is calculated for each district, with higher scores indicating greater readiness for digitally enabled SCF.


## Trade Finance Potential Score

Combine export orientation, import activity, and trade activity indicators to measure the district's **potential demand for trade finance within its supply chains**.


In [48]:
# Trade Finance Potential Score

trade_cols = [
    "Export_Potential_Score",
    "Import_Potential_Score",
    "Export_Activity_Score",
    "Import_Activity_Score"
]

df_final["Trade_Finance_Potential_Score"] = (
    df_final[trade_cols]
    .mean(axis=1)
    .round(2)
)

df_final[[
    "District",
    "Trade_Finance_Potential_Score"
]].head()

,District,Trade_Finance_Potential_Score
0,Abbottabad,1.0
1,Abbottabad,1.0
2,Abbottabad,1.0
3,Abbottabad,1.0
4,Abbottabad,1.0


### Outcome

A **Trade Finance Potential Score** is calculated for each district, with higher scores indicating greater trade-related financing potential.


## Final SCF Opportunity Score

Combine the key opportunity components using **weighted scoring** and normalize the result to a **0–100 scale** to create a single district-level SCF opportunity measure.


In [49]:
# Final Supply Chain Financing Opportunity Score

df_final["Supply_Chain_Financing_Opportunity_Score"] = (
    df_final["Supply_Chain_Potential_Score"] * 0.25 +
    df_final["Financing_Need_Score"] * 0.25 +
    df_final["Trade_Finance_Potential_Score"] * 0.20 +
    df_final["Digital_Readiness_Score"] * 0.15 +
    df_final["Export & Trade_Facilitation_Score"] * 0.15
) / 3 * 100

df_final["Supply_Chain_Financing_Opportunity_Score"] = (
    df_final["Supply_Chain_Financing_Opportunity_Score"]
    .round(2)
)
df_final[[
    "District",
    "Supply_Chain_Financing_Opportunity_Score"
]].sort_values(
    "Supply_Chain_Financing_Opportunity_Score",
    ascending=False
).head(10)

,District,Supply_Chain_Financing_Opportunity_Score
75,Gujranwala,80.05
74,Gujranwala,80.05
73,Gujranwala,80.05
72,Gujranwala,80.05
78,Gujranwala,80.05
79,Gujranwala,80.05
77,Gujranwala,80.05
76,Gujranwala,80.05
71,Gujranwala,80.05
70,Gujranwala,80.05


### Outcome

Each district receives a **Supply Chain Financing Opportunity Score from 0 to 100**, allowing districts to be compared based on their overall SCF potential.


## Opportunity Level

Classify districts into **Low, Medium, and High opportunity levels** based on their final SCF Opportunity Score to simplify district-level comparison.


In [50]:
# Opportunity Level

df_final["Opportunity_Level"] = pd.cut(
    df_final["Supply_Chain_Financing_Opportunity_Score"],
    bins=[0, 60, 80, 100],
    labels=["Low", "Medium", "High"],
    include_lowest=True
)

df_final[[
    "District",
    "Supply_Chain_Financing_Opportunity_Score",
    "Opportunity_Level"
]].head()

,District,Supply_Chain_Financing_Opportunity_Score,Opportunity_Level
0,Abbottabad,56.63,Low
1,Abbottabad,56.63,Low
2,Abbottabad,56.63,Low
3,Abbottabad,56.63,Low
4,Abbottabad,56.63,Low


### Outcome

Each district is classified into **Low, Medium, or High SCF opportunity** based on its score range.


In [51]:
df_final

,Year,Province,District,Key_Anchor_Sectors,Anchor_firm_presence,Avg_Employees_per_SME,Export_Oriented,Estimated_Number_of_SMEs,Estimated_Number_of_Employees,Banking_Penetration (%),...,Skilled_Labor_Availability_Score,Export & Trade_Facilitation_Score,Business_Environment_Score,Logistics_Score,Supply_Chain_Potential_Score,Financing_Need_Score,Digital_Readiness_Score,Trade_Finance_Potential_Score,Supply_Chain_Financing_Opportunity_Score,Opportunity_Level
0,2016,Khyber Pakhtunkhwa,Abbottabad,Hospitality & Tourism,Yes,9,No,7859,70731,55.0,...,3.0,1.0,2.14,3,1.83,1.94,2.71,1.0,56.63,Low
1,2017,Khyber Pakhtunkhwa,Abbottabad,Hospitality & Tourism,Yes,9,No,7941,71469,55.0,...,3.0,1.0,2.14,3,1.83,1.94,2.71,1.0,56.63,Low
2,2018,Khyber Pakhtunkhwa,Abbottabad,Hospitality & Tourism,Yes,10,No,8024,80240,55.0,...,3.0,1.0,2.14,3,1.83,1.94,2.71,1.0,56.63,Low
3,2019,Khyber Pakhtunkhwa,Abbottabad,Hospitality & Tourism,Yes,10,No,8108,81080,55.0,...,3.0,1.0,2.14,3,1.83,1.94,2.71,1.0,56.63,Low
4,2020,Khyber Pakhtunkhwa,Abbottabad,Hospitality & Tourism,Yes,9,No,8193,68575,55.0,...,3.0,1.0,2.14,3,1.83,1.94,2.71,1.0,56.63,Low
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
405,2021,Balochistan,Zhob,Livestock & Dairy,No,7,No,2025,13892,25.0,...,1.0,1.0,1.00,1,1.17,2.75,1.00,1.0,49.33,Low
406,2022,Balochistan,Zhob,Livestock & Dairy,No,8,No,2071,16568,25.0,...,1.0,1.0,1.00,1,1.17,2.75,1.00,1.0,49.33,Low
407,2023,Balochistan,Zhob,Livestock & Dairy,No,8,No,2119,17291,25.0,...,1.0,1.0,1.00,1,1.17,2.75,1.00,1.0,49.33,Low
408,2024,Balochistan,Zhob,Livestock & Dairy,No,8,No,2168,18038,25.0,...,1.0,1.0,1.00,1,1.17,2.75,1.00,1.0,49.33,Low


## Estimated Production Volume

Estimate district-level production volume using **SME scale, employment, production capacity, and processing capacity** as proxy indicators where direct production data is limited.


In [52]:
# Estimated Production Volume
# Logic: base scale from SMEs × employees, adjusted by capacity, anchor-firm, and employment scores

df_final['Estimated_Production_Volume'] = (
    df_final['Estimated_Number_of_SMEs'] *
    df_final['Avg_Employees_per_SME'] *
    df_final['SME_Scale_Score'] *
    df_final['Production_Capacity_Score'] *
    df_final['Processing_Capacity_Score'] *
    df_final['Employment_Scale_Score']
)

In [53]:
df_final[
    [
        'Province',
        'District',
        'Estimated_Number_of_SMEs',
        'Avg_Employees_per_SME',
        'SME_Scale_Score',
        'Production_Capacity_Score',
        'Processing_Capacity_Score',
        'Anchor_Firm_Score',
        'Employment_Scale_Score',
        'Estimated_Production_Volume'
    ]
]

,Province,District,Estimated_Number_of_SMEs,Avg_Employees_per_SME,SME_Scale_Score,Production_Capacity_Score,Processing_Capacity_Score,Anchor_Firm_Score,Employment_Scale_Score,Estimated_Production_Volume
0,Khyber Pakhtunkhwa,Abbottabad,7859,9,1,2,3,1,1,424386
1,Khyber Pakhtunkhwa,Abbottabad,7941,9,1,2,3,1,1,428814
2,Khyber Pakhtunkhwa,Abbottabad,8024,10,1,2,3,1,1,481440
3,Khyber Pakhtunkhwa,Abbottabad,8108,10,1,2,3,1,1,486480
4,Khyber Pakhtunkhwa,Abbottabad,8193,9,1,2,3,1,1,442422
...,...,...,...,...,...,...,...,...,...,...
405,Balochistan,Zhob,2025,7,1,1,3,0,1,42525
406,Balochistan,Zhob,2071,8,1,1,3,0,1,49704
407,Balochistan,Zhob,2119,8,1,1,3,0,1,50856
408,Balochistan,Zhob,2168,8,1,1,3,0,1,52032


### Outcome

An **Estimated Production Volume** is generated for each district to support comparison of production scale and SCF potential.


## Recommended Banking Product

SCF products are recommended using a **rule-based approach** that considers industry, value chain, anchor-firm presence, export orientation, banking penetration, and SME opportunity.

### Distributor / Channel Finance

Recommended for **retail, wholesale, and trading ecosystems** where financing is required across distributors, dealers, and sales channels.

### PO Finance

Recommended for **confirmed purchase-order based businesses**, particularly export-oriented manufacturers and suppliers requiring funds to fulfill orders.

### Payables Finance

Recommended where **anchor firms and established buyer–supplier relationships** exist, enabling financing of supplier invoices linked to anchor buyers.

### Receivables Discounting / Factoring

Recommended for businesses with **regular receivables and cash-flow cycles**, such as transport, logistics, professional services, and hospitality.

### Inventory Finance

Recommended for **inventory-intensive businesses** requiring working capital to purchase, hold, or process stock before sale.

### Debt-Based Financing

Recommended for **mining and mineral-related businesses** where financing requirements are more closely linked to capital-intensive operations.


In [54]:
def recommend_scf_product(row):

    sector = str(row["Key_Anchor_Sectors"]).lower()
    value_chain = str(row["Dominant_Value_Chain"]).lower()
    
    anchor = str(row["Anchor_firm_presence"]).lower() == "yes"
    export = str(row["Export_Oriented"]).lower() == "yes"
    
    banking = row["Banking_Penetration (%)"]
    opportunity = row["SME Opportunity_Score"]

    text = sector + " " + value_chain

    # 1. Distributor / Channel Finance
    if any(x in text for x in [
        "retail", "wholesale", "trade"
    ]):
        return "Distributor / Channel Finance"

    # 2. Manufacturing / Industrial Supply Chains
    if any(x in text for x in [
        "manufacturing", "textile", "apparel",
        "light engineering", "tobacco",
        "chemicals", "plastics"
    ]):
        
        if export and anchor:
            return "PO Finance + Payables Finance + Receivables Discounting"
        
        elif anchor:
            return "Payables Finance + Inventory Finance"
        
        else:
            return "Inventory Finance"

    # 3. Agriculture / Agri Supply Chains
    if any(x in text for x in [
        "agriculture", "agribusiness", "horticulture",
        "fruit", "rice", "sugarcane",
        "date palm", "livestock", "dairy",
        "fisheries"
    ]):
        
        if export:
            return "PO Finance + Inventory Finance"
        
        else:
            return "Inventory Finance"

    # 4. Mining / Minerals
    if any(x in text for x in [
        "mining", "minerals"
    ]):
        return "Debt‑Based Financing"

    # 5. Transport / Logistics
    if any(x in text for x in [
        "transport", "logistics"
    ]):
        return "Receivables Discounting / Factoring, PO Finance"

    # 6. IT / Technology Startups
    if any(x in sector for x in [
        "IT & Services",
        "Information Technology (IT)"
        
    ]):
        return "PO Finance"


    # 7. Professional Services / Hospitality / Tourism
    if any(x in text for x in [
        "professional services",
        "hospitality",
        "tourism"
    ]):
        return "Receivables Discounting / Factoring"


    # 8. General anchor-firm ecosystem
    if anchor and banking >= 70 and opportunity >= 30:
        return "Payables Finance"


    # 9. Default
        return "Inventory Finance"
        
df_final["Recommended_Banking_Product"] = df_final.apply(
    recommend_scf_product,
    axis=1
)

In [55]:
df_final

,Year,Province,District,Key_Anchor_Sectors,Anchor_firm_presence,Avg_Employees_per_SME,Export_Oriented,Estimated_Number_of_SMEs,Estimated_Number_of_Employees,Banking_Penetration (%),...,Business_Environment_Score,Logistics_Score,Supply_Chain_Potential_Score,Financing_Need_Score,Digital_Readiness_Score,Trade_Finance_Potential_Score,Supply_Chain_Financing_Opportunity_Score,Opportunity_Level,Estimated_Production_Volume,Recommended_Banking_Product
0,2016,Khyber Pakhtunkhwa,Abbottabad,Hospitality & Tourism,Yes,9,No,7859,70731,55.0,...,2.14,3,1.83,1.94,2.71,1.0,56.63,Low,424386,Receivables Discounting / Factoring
1,2017,Khyber Pakhtunkhwa,Abbottabad,Hospitality & Tourism,Yes,9,No,7941,71469,55.0,...,2.14,3,1.83,1.94,2.71,1.0,56.63,Low,428814,Receivables Discounting / Factoring
2,2018,Khyber Pakhtunkhwa,Abbottabad,Hospitality & Tourism,Yes,10,No,8024,80240,55.0,...,2.14,3,1.83,1.94,2.71,1.0,56.63,Low,481440,Receivables Discounting / Factoring
3,2019,Khyber Pakhtunkhwa,Abbottabad,Hospitality & Tourism,Yes,10,No,8108,81080,55.0,...,2.14,3,1.83,1.94,2.71,1.0,56.63,Low,486480,Receivables Discounting / Factoring
4,2020,Khyber Pakhtunkhwa,Abbottabad,Hospitality & Tourism,Yes,9,No,8193,68575,55.0,...,2.14,3,1.83,1.94,2.71,1.0,56.63,Low,442422,Receivables Discounting / Factoring
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
405,2021,Balochistan,Zhob,Livestock & Dairy,No,7,No,2025,13892,25.0,...,1.00,1,1.17,2.75,1.00,1.0,49.33,Low,42525,Inventory Finance
406,2022,Balochistan,Zhob,Livestock & Dairy,No,8,No,2071,16568,25.0,...,1.00,1,1.17,2.75,1.00,1.0,49.33,Low,49704,Inventory Finance
407,2023,Balochistan,Zhob,Livestock & Dairy,No,8,No,2119,17291,25.0,...,1.00,1,1.17,2.75,1.00,1.0,49.33,Low,50856,Inventory Finance
408,2024,Balochistan,Zhob,Livestock & Dairy,No,8,No,2168,18038,25.0,...,1.00,1,1.17,2.75,1.00,1.0,49.33,Low,52032,Inventory Finance


In [56]:
columns = df_final.columns.tolist()

print(columns)

['Year', 'Province', 'District', 'Key_Anchor_Sectors', 'Anchor_firm_presence', 'Avg_Employees_per_SME', 'Export_Oriented', 'Estimated_Number_of_SMEs', 'Estimated_Number_of_Employees', 'Banking_Penetration (%)', 'SME Opportunity_Score', 'Dominant_Value_Chain', 'Dominant_Industry', 'Infrastructure (Roads, Electricity, Internet)', 'Banking_Access', 'Access_to_Credit / Lending', 'Business_Registration', 'Tax_System', 'Property_Registration', 'Contract_Enforcement', 'Legal & Regulatory_Environment', 'Skilled_Labor_Availability', 'Export & Trade_Facilitation', 'Digital_payment_Adoption', 'Logistics/Transport_Connectivity', 'Supply_Chain_Constraints', 'Urbanization_Level', 'Production_Volume', 'Processing_Capacity', 'Export_Activity', 'Import_Activity', 'SME_Scale_Score', 'Employment_Scale_Score', 'Anchor_Firm_Score', 'Production_Capacity_Score', 'Processing_Capacity_Score', 'Credit_Access_Score', 'Banking_Access_Score', 'Supply_Chain_Constraint_Score', 'Working_Capital_Gap_Score', 'Export_Po

## Select Final SCF Variables

Select the **key analytical and business variables** required for the final SCF dataset, keeping the dataset concise and suitable for **Power BI visualization and management reporting**. 

Further columns will also be added in future if needed.



In [57]:
sc_columns = [
    'Year',
    'Province',
    'District',
    'Dominant_Industry',
    'Key_Anchor_Sectors',
    'Anchor_firm_presence',
    'Dominant_Value_Chain',
    'Estimated_Number_of_SMEs',
    'Estimated_Number_of_Employees',
    'Production_Volume',
    'Processing_Capacity',
    'Infrastructure (Roads, Electricity, Internet)',
    'Logistics/Transport_Connectivity',
    'Banking_Penetration (%)',
    'Access_to_Credit / Lending',
    'Working_Capital_Gap_Score',
    'Estimated_Production_Volume',
    'Export_Oriented',
    'Supply_Chain_Financing_Opportunity_Score',
    'Opportunity_Level',
    'Recommended_Banking_Product'
]
df_final = df_final[sc_columns]


# Check
print("Shape:", df_final.shape)
print("Columns:")
print(df_final.columns.tolist())

Shape: (410, 21)
Columns:
['Year', 'Province', 'District', 'Dominant_Industry', 'Key_Anchor_Sectors', 'Anchor_firm_presence', 'Dominant_Value_Chain', 'Estimated_Number_of_SMEs', 'Estimated_Number_of_Employees', 'Production_Volume', 'Processing_Capacity', 'Infrastructure (Roads, Electricity, Internet)', 'Logistics/Transport_Connectivity', 'Banking_Penetration (%)', 'Access_to_Credit / Lending', 'Working_Capital_Gap_Score', 'Estimated_Production_Volume', 'Export_Oriented', 'Supply_Chain_Financing_Opportunity_Score', 'Opportunity_Level', 'Recommended_Banking_Product']


In [58]:
df_final

,Year,Province,District,Dominant_Industry,Key_Anchor_Sectors,Anchor_firm_presence,Dominant_Value_Chain,Estimated_Number_of_SMEs,Estimated_Number_of_Employees,Production_Volume,...,"Infrastructure (Roads, Electricity, Internet)",Logistics/Transport_Connectivity,Banking_Penetration (%),Access_to_Credit / Lending,Working_Capital_Gap_Score,Estimated_Production_Volume,Export_Oriented,Supply_Chain_Financing_Opportunity_Score,Opportunity_Level,Recommended_Banking_Product
0,2016,Khyber Pakhtunkhwa,Abbottabad,Hospitality & Tourism,Hospitality & Tourism,Yes,Cotton,7859,70731,Medium,...,High,High,55.0,Medium,1.75,424386,No,56.63,Low,Receivables Discounting / Factoring
1,2017,Khyber Pakhtunkhwa,Abbottabad,Hospitality & Tourism,Hospitality & Tourism,Yes,Cotton,7941,71469,Medium,...,High,High,55.0,Medium,1.75,428814,No,56.63,Low,Receivables Discounting / Factoring
2,2018,Khyber Pakhtunkhwa,Abbottabad,Hospitality & Tourism,Hospitality & Tourism,Yes,Cotton,8024,80240,Medium,...,High,High,55.0,Medium,1.75,481440,No,56.63,Low,Receivables Discounting / Factoring
3,2019,Khyber Pakhtunkhwa,Abbottabad,Hospitality & Tourism,Hospitality & Tourism,Yes,Cotton,8108,81080,Medium,...,High,High,55.0,Medium,1.75,486480,No,56.63,Low,Receivables Discounting / Factoring
4,2020,Khyber Pakhtunkhwa,Abbottabad,Hospitality & Tourism,Hospitality & Tourism,Yes,Cotton,8193,68575,Medium,...,High,High,55.0,Medium,1.75,442422,No,56.63,Low,Receivables Discounting / Factoring
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
405,2021,Balochistan,Zhob,Livestock & Dairy,Livestock & Dairy,No,Livestock & Dairy,2025,13892,Low,...,Low,Low,25.0,Low,2.00,42525,No,49.33,Low,Inventory Finance
406,2022,Balochistan,Zhob,Livestock & Dairy,Livestock & Dairy,No,Livestock & Dairy,2071,16568,Low,...,Low,Low,25.0,Low,2.00,49704,No,49.33,Low,Inventory Finance
407,2023,Balochistan,Zhob,Livestock & Dairy,Livestock & Dairy,No,Livestock & Dairy,2119,17291,Low,...,Low,Low,25.0,Low,2.00,50856,No,49.33,Low,Inventory Finance
408,2024,Balochistan,Zhob,Livestock & Dairy,Livestock & Dairy,No,Livestock & Dairy,2168,18038,Low,...,Low,Low,25.0,Low,2.00,52032,No,49.33,Low,Inventory Finance


### Outcome

The final dataset is reduced to **21 relevant columns**, containing the core district, SME, supply-chain, financing, trade, opportunity, and product recommendation variables.


## Normalize Working Capital Gap Score

Convert the **Working Capital Gap Score from a 1–3 scale to a 0–100 scale** for consistent comparison with other SCF opportunity indicators.


In [59]:
df_final["Working_Capital_Gap_Score"] = (
    (df_final["Working_Capital_Gap_Score"] / 3) * 100
).round(2)

### Outcome

The Working Capital Gap Score is now expressed on a **0–100 scale**, making it easier to compare across districts and integrate into the final SCF analysis.


## Export Final Dataset

Export the finalized SCF opportunity dataset as a **CSV file** for use in Power BI and further reporting.
 

In [60]:
# Export df_final to CSV
df_final.to_csv("SC_Opportunity_Data 1.csv", index=False, encoding="utf-8-sig")

print("CSV exported successfully!")
print("File: SC_Opportunity_Data.csv")

CSV exported successfully!
File: SC_Opportunity_Data.csv


### Outcome

The final SCF dataset is successfully exported as a **CSV file**, ready for Power BI visualization.


## EDA & Visualization in Power BI

Further **Exploratory Data Analysis (EDA), interactive visualizations, and dashboard development** were conducted in **Power BI** using the finalized CSV dataset.
